### PR TITLE
Explore: Refactor graph component to use PanelRenderer

### LINKED ISSUE #38866: Refactor Explore's graph to use PanelRenderer instead of custom ExploreGraphNG
Explore currently uses [custom graph panel implementation](https://github.com/grafana/grafana/blob/main/public/app/features/explore/ExploreGraphNGPanel.tsx)which is pretty much the same implementation as TimeSeries panel. 

Let's rewrite it to use PanelRenderer with TimeSeriesPanel rather than having this additional wrapper. PanelRenderer with TimeSeriesPanel is already used successfully with Alerting NG, i..e https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/rule-viewer/RuleViewerVisualization.tsx#L110

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

In this PR I refactored explore's graph component to use the Panelrenderer component instead of the TimeSeries component directly. I'm not sure if we want to make a more generic component to render panels in Explore?

**Which issue(s) this PR fixes**:

Fixes #38866

### CODE REVIEW

File: packages/grafana-ui/src/components/PanelChrome/PanelContext.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ export interface PanelContext {
   onAnnotationCreate?: (annotation: AnnotationEventUIModel) => void;
   onAnnotationUpdate?: (annotation: AnnotationEventUIModel) => void;
   onAnnotationDelete?: (id: string) => void;
+  /**
+   * splitOpenFn is used in Explore to open the split view. It can be used in panels which has intercations and used in Explore as well.
+   * For example TimeSeries panel.
+   */
+  splitOpenFn?: SplitOpen;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:29]:
```suggestion
  onSplitOpen?: SplitOpenFn;
```

File: public/app/features/explore/ExploreGraph.tsx [hunk 2]
```diff
@@ -6,63 +7,58 @@ import {
...
   width: number;
-  annotations?: DataFrame[];
   absoluteRange: AbsoluteTimeRange;
   timeZone: TimeZone;
-  onUpdateTimeRange: (absoluteRange: AbsoluteTimeRange) => void;
+  loadingState: LoadingState;
+  annotations?: DataFrame[];
   onHiddenSeriesChanged?: (hiddenSeries: string[]) => void;
-  tooltipDisplayMode: TooltipDisplayMode;
+  tooltipDisplayMode?: TooltipDisplayMode;
```
Comment (dprokop) [inline public/app/features/explore/ExploreGraph.tsx:47]: I don't think this property is needed as it's fixed (Multi) anyways from what I see.

Comment (dprokop) [reply public/app/features/explore/ExploreGraph.tsx:47]: My bad- I see it differs between logs and series.

**Review by dprokop — APPROVED** [review]
Think this looks good, awesome work @zoltanbedi 👍 

I'll leave the final call to @ifrost and @aocenas.

**Review by aocenas — APPROVED** [review]

File: packages/grafana-ui/src/components/PanelChrome/PanelContext.ts (hunk 1 above)
Comment (zoltanbedi) [reply packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:29]: would it make sense if I'd only rename the property and not the type?

Comment (dprokop) [reply packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:29]: yes

**Comment by ifrost** [comment]
Did it break zooming in the time range in logs histogram? 🤔  @zoltanbedi

**Comment by zoltanbedi** [comment]
> Did it break zooming in the time range in logs histogram? 🤔 @zoltanbedi

I don't know.

**Pull request merged**